### PR TITLE
unionstore: add checkpoint API to support TiDB savepoint feature

### PR DESCRIPTION
**Pull request by crazycs520**

### CODE REVIEW

**Review by you06 — APPROVED** [review]

File: internal/unionstore/memdb.go [hunk 1]
```diff
@@ -155,6 +155,20 @@ func (db *MemDB) Cleanup(h int) {
 	db.stages = db.stages[:h-1]
 }
 
+type MemCheckpoint = memdbCheckpoint
```
Comment (bb7133) [inline internal/unionstore/memdb.go:158]: How about changing `memdbCheckpoint` to public?

**Comment by bb7133** [comment]
Please sign the DCO.

File: internal/unionstore/memdb.go (hunk 1 above)
Comment (crazycs520) [reply internal/unionstore/memdb.go]: ok, done.

**Pull request merged**